### ISSUE TITLE
pdfjs has a security vulnerability under version 3.2.146. Can it be upgraded to 4.2.67 or higher?

### ISSUE THREAD

**Issue by ava-zhang1** [react-pdf-viewer/react-pdf-viewer#1874]
Affected versions of this package are vulnerable to Arbitrary Code Injection in font_loader.js, which passes input to the eval() function when the default isEvalSupported option is in use. An attacker can execute code by convincing a user to open a malicious PDF file.

**Comment by godehk** [comment]
We are also getting same below  , could you please fix asap and we bought license too

**If pdf.js is used to load a malicious PDF, and PDF.js is configured with isEvalSupported set to true (which is the default value), unrestricted attacker-controlled JavaScript will be executed in the context of the hosting domain.**

**Evidence**
Vulnerable Package
pdfjs-dist:3.11.174

**Solution:**
Upgrade to version 4.2.67 or above.

**Comment by alessiofnes** [comment]
> We are also getting same below , could you please fix asap and we bought license too

I agree, any updates?

**Comment by joebaileyrsm** [comment]
Agreed, We need update on this, it is obviously impacting multiple customers.

**Comment by liangskyli** [comment]
I also encountered the same problem and hope to upgrade and update

**Comment by shaangidwani** [comment]
I have same issue any timeline to update this package as we are stuck on this.

**Comment by olhapi** [comment]
You can pass this to your Viewer component to disable JS in PDF `transformGetDocumentParams={(params) => ({ ...params, isEvalSupported: false })}`

**Comment by david-reactpdf** [comment]
@ava-zhang1 @godehk @alessiofnes @joebaileyrsm @liangskyli @shaangidwani 

Hi all, just wanted to share in case it’s helpful.

If you’re evaluating alternatives and need something actively maintained with up-to-date PDF.js support, you might want to take a look at our library React PDF: [https://www.react-pdf.dev](https://www.react-pdf.dev)

It’s a paid library, but we ship regular updates and are currently running on the latest PDF.js v4. We also allow overriding pdfjs-dist if your security policy requires a specific version. In addition, PDF.js v5 support is already in progress and will be rolling out soon.

Feel free to reach out with any questions if you’re exploring options.

**Comment by LazyCompiler** [comment]
For anyone researching alternatives, we recently evaluated several PDF viewer solutions and ultimately chose https://www.embedpdf.com/.

We did consider react-pdf.dev as well, but we preferred EmbedPDF because it can be integrated without wrapping the app (or large subtrees) in additional config/context providers. It's also 100% free and open source, which was an important factor for us.

We've also been very pleased with the performance of Pdfium compared to PDF.js.

So far, it's been a great fit for our needs.
